src/Compilers/CSharp/Portable/Binder/Binder_Initializers.cs

```diff
@@ -115,6 +115,17 @@ internal struct ProcessedFieldInitializers
                                 parentBinder = parentBinder.GetFieldInitializerBinder(fieldSymbol);
 
                                 BoundFieldEqualsValue boundInitializer = BindFieldInitializer(parentBinder, fieldSymbol, initializerNode, diagnostics);
+
+                                if (!boundInitializer.Value.HasAnyErrors)
+                                {
+                                    var field = boundInitializer.Field;
+                                    if (field.Type.IsRefLikeType)
+                                    {
+                                        var value = parentBinder.ValidateEscape(boundInitializer.Value, ExternalScope, isByRef: false, diagnostics);
```

Comment:
>ValidateEscape [](http://example.com/codeflow?start=65&length=14)

We don't need to do the same check in ```BindScriptFieldInitializers```?

Reply:
`IsRefLikeType` fields cannot be declared in a script since the containing type is not a `ref struct`. Added a test.